banzai/astrometry.py

```diff
@@ -32,6 +32,49 @@ def do_stage(self, image):
             image.header['WCSERR'] = (4, 'Error status of WCS fit. 0 for no error')
             return image
 
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            filename = os.path.basename(image.filename)
+            catalog_name = os.path.join(tmpdirname, filename.replace('.fits', '.cat.fits'))
+            try:
+                image.write_catalog(catalog_name, nsources=40)
+            except MissingCatalogException:
+                image.header['WCSERR'] = (4, 'Error status of WCS fit. 0 for no error')
+                logger.error('No source catalog. Not attempting WCS solution', image=image)
+                return image
+
+            # Run astrometry.net
+            wcs_name = os.path.join(tmpdirname, filename.replace('.fits', '.wcs.fits'))
+            command = self.cmd.format(ra=image.ra, dec=image.dec, scale_low=0.9*image.pixel_scale,
+                                      scale_high=1.1*image.pixel_scale, wcs_name=wcs_name,
+                                      catalog_name=catalog_name, nx=image.nx, ny=image.ny)
+
+            try:
+                console_output = subprocess.check_output(shlex.split(command))
+            except subprocess.CalledProcessError:
+                logger.error('Astrometry.net threw an error.', image=image)
+
+            logger.debug(console_output, image=image)
+
+            if os.path.exists(wcs_name):
+                # Copy the WCS keywords into original image
+                new_header = fits.getheader(wcs_name)
+
+                header_keywords_to_update = ['CTYPE1', 'CTYPE2', 'CRPIX1', 'CRPIX2', 'CRVAL1',
+                                             'CRVAL2', 'CD1_1', 'CD1_2', 'CD2_1', 'CD2_2']
+                for keyword in header_keywords_to_update:
+                    image.header[keyword] = new_header[keyword]
+
+                image.header['WCSERR'] = (0, 'Error status of WCS fit. 0 for no error')
+
+                # Update the RA and Dec header keywords
+                image.header['RA'], image.header['DEC'] = get_ra_dec_in_sexagesimal(image.header['CRVAL1'],
+                                                                                    image.header['CRVAL2'])
+
+                # Clean up wcs file
+                os.remove(wcs_name)
+
+                # Add the RA and Dec values to the catalog
+                add_ra_dec_to_catalog(image)
```

Comment:
Was the merge with the GAIA changes done correctly? I wouldn't expect any major updates to `astrometry.py`